tutorials/haskell/hspec-golden-test-tutorial/tutorial.md

```diff
@@ -0,0 +1,506 @@
+---
+
+
+The importance of software testing relies on the ability to check if a determined application behaves in an appropriate way. Unit tests are the first level of testing and helps developers to detect early bugs in the code by testing each component individually.
+
+For example, lets create a function that welcomes someone to this tutorial:
+
+```haskell
+sayHi :: String -> String
+sayHi name = "Welcome to the Golden Tests tutorial " ++ name
+```
+
+Using ```hspec``` we can easily test our function :
+
+```haskell
+import HelloWorld
+import Test.Hspec
+
+spec :: Spec
+spec = do
+    describe "sayHi" $ 
+        it "shows a Hello Golden Testers string" $
+            (sayHi "John Doe") `shouldBe` "Welcome to the Golden Tests tutorial John Doe"
+```
+
+So, in a nutshell we are comparing the output of the function with a string that is stored inside the code of our test. We can take a different approach and store this expected output in a separate file. This approach is known as golden testing and the file in which we store the expected output takes the name of "golden file".
+
+So, now we know what golden testing is, but if unit tests and golden tests are so similar, why don't we just keep up with unit tests, right?
+
+
+The truth is that unit tests are not useful when evaluating complex and large outputs and that's where the difference lies. Test are all about maintainability and readability and when we tests large outputs we tend to add visual noise to our code. Helpfully Golden Tests fixes this issue, lets check the main advantages.
+
+ - Large outputs are stored in a golden file reducing the visual noise in our code.
+
+ - With some golden tests libraries the expected output (Golden File) can be automatically generated.
+
+ - Some golden test libraries also automatically updates the golden file. So, we don't need to manually update the tests when the expected output changes. 
+
+ - It is useful when working with JSON, HTML, images, etc.
+
+In this tutorial we will use [hspec-golden](http://hackage.haskell.org/package/hspec-golden) library to build golden tests. We will also use the ```hspec-golden``` CLI to easily update them.
+
+You can either code along with this tutorial or check at the finished [code](https://github.com/stackbuilders/tutorials/tree/tutorials/tutorials/haskell/hspec-golden-test-tutorial/code) in Github. 
+
+## Hspec-golden
+
+`Hspec-golden` is a StackBuilders testing library, written in haskell, that helps users implement golden tests. It is an OpenSource project so feel free to [review](https://github.com/stackbuilders/hspec-golden) it.
+
+## Getting Started
+
+First things first, lets create a new Haskell project on which we will be working:
+
+```bash
+stack new "hspec-golden-tests"
```

Comment:
@Jagl257 I think the name of the example project change to `hspec-golden-test-tutorial`. Could you please change it here?